src/clib/lib/enkf/obs_vector.cpp

```diff
@@ -81,29 +99,22 @@ __conf_instance_get_restart_nr(const conf_instance_type *conf_instance,
             conf_instance_get_item_value_double(conf_instance, "HOURS");
         util_inplace_forward_seconds_utc(&obs_time, hours * 3600);
     } else
-        util_abort("%s: Internal error. Invalid conf_instance?\n", __func__);
-
-    auto it = std::lower_bound(time_map.begin(), time_map.end(), obs_time);
-    const time_t tolerance = 30;
-    if (it == time_map.end() || std::abs(obs_time - *it) >= tolerance) {
+        throw std::logic_error(fmt::format(
+            "Observation: {} missing DATE, DAYS or HOURS", obs_key));
+    obs_restart_nr = find_within_tolerance(time_map, obs_time);
+    if (obs_restart_nr < 0) {
+        std::string error = fmt::format(
```

Comment:
I personally prefer to not split up exception message strings unless absolutely necessary. But it's personal taste so up to you.

Reply:
It was just a lot of repretition if this was not done, and a blanket if someone is using `HOURS`, which may or may not be possible